### PR TITLE
Describe `tf.squeeze`'s None shape behavior.

### PR DESCRIPTION
**Pull request by rainwoodman**

### CODE REVIEW

File: tensorflow/python/ops/array_ops.py [hunk 1]
```diff
@@ -4622,12 +4622,36 @@ def squeeze_v2(input, axis=None, name=None):
...
-
+  
   Unlike the older op `tf.compat.v1.squeeze`, this op does not accept a
   deprecated `squeeze_dims` argument.
 
   Note: if `input` is a `tf.RaggedTensor`, then this operation takes `O(N)`
   time, where `N` is the number of elements in the squeezed dimensions.
+  
+  Note: If squeeze is performed on dimensions of unknown sizes, then the
+  returned   Tensor will be of unknown shape. In Keras the first (batch)
```
Comment (rohan100jain) [inline tensorflow/python/ops/array_ops.py:4633]: Please avoid mentioning Keras in low level TF API documentation. I think you can mention that in general the first dimension of Tensors using in ML training workloads is an unknown batch dimension.

**Review by rohan100jain — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: tensorflow/python/ops/array_ops.py (hunk 1 above)
Comment (rainwoodman) [reply tensorflow/python/ops/array_ops.py]: Done.

File: tensorflow/python/ops/array_ops.py [hunk 2]
```diff
@@ -4622,12 +4622,36 @@ def squeeze_v2(input, axis=None, name=None):
...
+    y = tf.squeeze(x, axis=known_axes)
+    print('shape of tf.squeeze(x, axis=known_axes):', y.shape)
+    y = tf.squeeze(x)
+    print('shape of tf.squeeze(x):', y.shape)
+    return 0
+ 
+  _ = func.get_concrete_function(tf.TensorSpec([None, 1, 2], dtype=tf.int32))
+  # Output is.
+  # x.shape: (None, 1, 2)
+  # shape of tf.squeeze(x, axis=known_axes): (None, 2)
```
Comment (rohan100jain) [inline tensorflow/python/ops/array_ops.py:4652]: nit: You might want to actually print out `known_axes` i.e. [1] here?

**Review by rohan100jain — APPROVED** [review]

**Pull request merged**